examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py

```diff
@@ -571,7 +636,7 @@ def parse_args(input_args=None):
     parser.add_argument(
         "--optimizer",
         type=str,
-        default="adamW",
+        default="AdamW",
         help=('The optimizer type to use. Choose between ["AdamW", "prodigy"]'),
```

Comment:
you can set this like:

```py
available_optimizers = ['AdamW', 'prodigy']
parser.add_argument(
...
    choices=available_optimizers,
    help=f'The optimizer type to use. Choose between {available_optimizers}'
)
```

Reply:
which ensures the user doesn't select a nonexistent option

Reply:
Yes, +1 to this suggestion. `choices` greatly simplifies things.

Reply:
That would be a breaking change though, because atm - if a user specifies an optimizer other than Adam/prodigy, we log a warning default to Adam (see below)

```
# Optimizer creation
    if not (args.optimizer.lower() == "prodigy" or args.optimizer.lower() == "adamw"):
        logger.warning(
            f"Unsupported choice of optimizer: {args.optimizer}.Supported optimizers include [adamW, prodigy]."
            "Defaulting to adamW"
        )
        args.optimizer = "adamw"
```